statsprinter.go

```diff
@@ -255,95 +256,174 @@ JSON output section
 
 */
 
-/* Print message  in JSON format */
-func jsonPrintf(format string, a ...interface{}) {
-	data := struct {
-		Message string `json:"message"`
-	}{
-		Message: fmt.Sprintf(format, a...),
-	}
-	outputJson, _ := json.Marshal(&data)
-	fmt.Println(string(outputJson))
+type JSONEventType string
+
+const (
+	// Probe is an event type that represent 1 probe / ping / request event.
+	Probe JSONEventType = "probe"
+	// Retry is an event type that's being sent,
+	// when tcping retries to resolve a hostname.
+	Retry JSONEventType = "retry"
+	// RetrySuccess is a sub-type for of Retry event,
+	// when previous retry was unsuccessful, but the next one became successful.
+	RetrySuccess JSONEventType = "retry-success"
+	// Start is an event type that's sent only one,
+	// before doing any actual work.
+	Start JSONEventType = "start"
+	// Stats is a final event that's being sent
+	// before tcping exits.
+	Stats JSONEventType = "stats"
```

Comment:
Any particular reason for exporting these?

Reply:
No, not really.

I accidentally marked this as resolved because I thought it's the same case as in JSONData, but no, these could be unexported, and it shouldn't affect anything.

Reply:
Ok, let's "un-export" them for now.
In the future we can export them if need be.

Reply:
By the way, I hope you are not tired of my comments 😄 

I will surely try to finalize this review this Saturday. These days I am a bit tied up with other stuff.

Cheers

Reply:
> By the way, I hope you are not tired of my comments 😄

Nah, all good, don't worry 👍

Reply:
Let's not export these.

Reply:
I did fix it, didn't I? Or you the `JSONEventType` as well?

Reply:
Strange. Maybe a caching issue on my side. It is fine now.